classes/03class/exercises/c03-aws02/nimmi89/README.md

```diff
@@ -0,0 +1,206 @@
+# c03-aws02
+
+## Command Execution Output
+- Command used to create the Application Load Balancer (ALB)
+```
+> Command to create load balancer
+aws elbv2 create-load-balancer --name my-load-balancer  \
+--subnets subnet-062c79f06460c35c1 subnet-07debbb9f04cb105c --security-groups sg-005dc45ad043be6c1
+{
+    "LoadBalancers": [
+        {
+            "LoadBalancerArn": "arn:aws:elasticloadbalancing:ap-southeast-2:438549961569:loadbalancer/app/my-load-balancer/c73a2084dae67ce8",
+            "DNSName": "my-load-balancer-203056255.ap-southeast-2.elb.amazonaws.com",
+            "CanonicalHostedZoneId": "Z1GM3OXH4ZPM65",
+            "CreatedTime": "2020-07-11T11:23:54.500000+00:00",
+            "LoadBalancerName": "my-load-balancer",
+            "Scheme": "internet-facing",
+            "VpcId": "vpc-081afcee350babdf5",
+            "State": {
+                "Code": "provisioning"
+            },
+            "Type": "application",
+            "AvailabilityZones": [
+                {
+                    "ZoneName": "ap-southeast-2a",
+                    "SubnetId": "subnet-062c79f06460c35c1",
+                    "LoadBalancerAddresses": []
+                },
+                {
+                    "ZoneName": "ap-southeast-2b",
+                    "SubnetId": "subnet-07debbb9f04cb105c",
+                    "LoadBalancerAddresses": []
+                }
+            ],
+            "SecurityGroups": [
+                "sg-005dc45ad043be6c1"
+            ],
+            "IpAddressType": "ipv4"
+        }
+    ]
+}
+
+ > Command to create target group
+aws elbv2 create-target-group \
+    --name DA-targets --protocol HTTP \
+    --port 80 \
+    --target-type instance \
+    --vpc-id vpc-081afcee350babdf5
+{
+    "TargetGroups": [
+        {
+            "TargetGroupArn": "arn:aws:elasticloadbalancing:ap-southeast-2:438549961569:targetgroup/DA-targets/95c890365deb9657",
+            "TargetGroupName": "DA-targets",
+            "Protocol": "HTTP",
+            "Port": 80,
+            "VpcId": "vpc-081afcee350babdf5",
+            "HealthCheckProtocol": "HTTP",
+            "HealthCheckPort": "traffic-port",
+            "HealthCheckEnabled": true,
+            "HealthCheckIntervalSeconds": 30,
+            "HealthCheckTimeoutSeconds": 5,
+            "HealthyThresholdCount": 5,
+            "UnhealthyThresholdCount": 2,
+            "HealthCheckPath": "/",
+            "Matcher": {
+                "HttpCode": "200"
+            },
+            "TargetType": "instance"
+        }
+    ]
+}
+
+> Command to register your instances with your target group
+aws elbv2 register-targets \
+    --target-group-arn arn:aws:elasticloadbalancing:ap-southeast-2:438549961569:targetgroup/DA-targets/95c890365deb9657 \
+    --targets Id=i-073c18d845b092950 Id=i-09b699355152b2618
```

Comment:
What happens if one of your instances goes down?

Probably the ASG will scale up another one to reach the desired state. But will it be registered and reachable via 
load balancer? (No)

For auto (de)registering instances created by the ASG you need to associate the ASG with the Target group:

```
aws autoscaling attach-load-balancer-target-groups \
    --auto-scaling-group-name devops-academy-asg \
    --target-group-arns arn:aws:elasticloadbalancing:ap-southeast-2:149613515908:targetgroup/devops-academy-tg/16871e3110304536
```

Then you don't need to register specific instance IDs in your load balancer

Reply:
As the exercise didn't request it, it is all cool :)